[Conversation API] initial docs setup (#4419)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/conversation/conversation-overview.md

---
type: docs
title: "Conversation overview"
linkTitle: "Overview"
weight: 1000
description: "Overview of the conversation API building block"
---

> Using the Dapr conversation API you can reduce the complexity of interacting with LLMS and enable critical security and reliability functionality with out-of-the-box features like prompt caching and PII data obfuscation.

Reviewer: ```suggestion
Using the Dapr conversation API you can reduce the complexity of interacting with Large Language Models (LLMs) and enable critical performance and security functionality with features like prompt caching and PII data obfuscation.
```

## Features

> 

Reviewer: Add these sub-titles
### Prompt caching
TBD

### PII obfuscation
TBD

> ## Try out pub/sub

Reviewer: ```suggestion
## Try out conversation
```

### Quickstarts and tutorials

Want to put the Dapr conversation API to the test? Walk through the following quickstart and tutorials to see it in action:

| Quickstart/tutorial | Description |
| ------------------- | ----------- |
| [Conversation quickstart](todo) | . |

### Start using pub/sub directly in your app

Want to skip the quickstarts? Not a problem. You can try out the conversation building block directly in your application. After [Dapr is installed]({{< ref "getting-started/_index.md" >}}), you can begin using the conversation API starting with [the how-to guide](todo).

## Next steps

> - [How-To: Migrate between large language models]({{< ref howto-conversation-layer.md >}})

Reviewer: This needs updating

---
type: docs
title: "Conversation overview"
linkTitle: "Overview"
weight: 1000
description: "Overview of the conversation API building block"
---

{{% alert title="Alpha" color="primary" %}}
The conversation API is currently in [alpha]({{< ref "certification-lifecycle.md#certification-levels" >}}).
{{% /alert %}}

Using the Dapr conversation API, you can reduce the complexity of interacting with Large Language Models (LLMs) and enable critical performance and security functionality with features like prompt caching and personally identifiable information (PII) data obfuscation.

## Features

### Prompt caching

> todo

Reviewer: ```suggestion
Prompt caching stores frequently prompt in a cache to reuse this cached information instead of reprocessing it for every new request, which significantly reduces latency and cost. Essentially, it's a way to optimize performance by storing and reusing prompts that are often repeated across multiple API calls.
```

### Personally identifiable information (PII) obfuscation

> todo

Reviewer: ```suggestion
PII obfuscation identifies and removes any personally identifiable information (PII) from a conversation response essentially cleaning the data by eliminating sensitive details like names, addresses, phone numbers or other details that could be used to identify an individual, thus protecting privacy.
```

> ## Supported SDKs

Reviewer: I suggest removing this section and just have the quickstart links.

You can use the following SDKs to author a workflow.

| Language stack | Package |
| - | - |
| .NET | [Dapr.AI.Conversation](todo) |
| Rust | [converse](https://github.com/dapr/rust-sdk/blob/main/dapr/src/client.rs#L542) |
| Go | [conversation](https://github.com/dapr/go-sdk/tree/main/client/conversation.go) |

## Try out conversation

### Quickstarts and tutorials

Want to put the Dapr conversation API to the test? Walk through the following quickstart and tutorials to see it in action:

| Quickstart/tutorial | Description |
| ------------------- | ----------- |
| [Conversation quickstart](todo) | . |

> ### Start using pub/sub directly in your app

Reviewer: ```suggestion
### Start using conversation API in your app
```

> Want to skip the quickstarts? Not a problem. You can try out the conversation building block directly in your application. After [Dapr is installed]({{< ref "getting-started/_index.md" >}}), you can begin using the conversation API starting with [the how-to guide](todo).

Reviewer: Update the link to the HowTo guide with the one in the next steps
